src/states/common/approved.rs

```diff
@@ -316,9 +316,23 @@ pub trait Approved: Base {
             our_pfx = *self.chain_mut().our_prefix();
         }
 
+        self.check_voting_status();
+
         Ok(Transition::Stay)
     }
 
+    // Checking members vote status and vote to remove those non-resposive nodes.
+    fn check_voting_status(&mut self) {
+        let unresponsive_nodes = self.chain_mut().check_vote_status();
+        let log_indent = self.log_ident();
+        for pub_id in unresponsive_nodes.iter() {
+            self.parsec_map_mut().vote_for(
+                AccumulatingEvent::RemoveElder(*pub_id).into_network_event(),
```

Comment:
1 - Should we be voting Offline?
2 - Do we need to remove `self.vote_for_event(AccumulatingEvent::Offline(pub_id));` from handle_peer_lost?

Reply:
> Should we be voting Offline
The `handle_offline` doesn't remove the peer from the elders. However we need to do so as this is an expected voter (which is supposed to be an elder) to be removed?

> Do we need to remove self.vote_for_event(AccumulatingEvent::Offline(pub_id)); from handle_peer_lost?
Maybe not, as it is still required for the non-elder peer? If the peer is an elder, the duplicated vote of Offline (we also cast Offline vote inside handle_remove_elder) shall be filtered.

Reply:
> The handle_offline doesn't remove the peer from the elders

It calls remove_member which does vote to remove the elder.

> Maybe not, as it is still required for the non-elder peer? 

Possibly.

Reply:
> It calls remove_member which does vote to remove the elder.

Ah, right, I messed it with the one in adult. Will change it to vote Offline then.